File: Examples/VisualBasic/ProgrammersGuide/Working-Shapes/ReadShapePropByName.vb
Imports VisualBasic
Imports Aspose.Diagram
Imports System

Public Class ReadShapePropByName
    Public Shared Sub Run()
        'ExStart:ReadShapePropByName
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_Shapes()

        ' call a Diagram class constructor to load the VSDX diagram
        Dim diagram As New Diagram(dataDir & Convert.ToString("Drawing1.vsdx"))
        ' get page by name
        Dim page As Page = diagram.Pages.GetPage("Page-3")

        For Each shape As Aspose.Diagram.Shape In page.Shapes
            If shape.Name = "Process1" Then
                Dim [property] As Prop = shape.Props.GetProp("Name1")
                Console.WriteLine([property].Label.Value + ": " + [property].Value.Val)
            End If
        Next
        'ExEnd:ReadShapePropByName
    End Sub
End Class

File: Examples/CSharp/ProgrammersGuide/Working-Shapes/ReadShapePropByName.cs
using Aspose.Diagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CSharp.ProgrammersGuide.Working_Shapes
{
    public class ReadShapePropByName
    {
        public static void Run()
        {
            //ExStart:ReadShapePropByName
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Shapes();

            // call a Diagram class constructor to load the VSDX diagram
            Diagram diagram = new Diagram(dataDir + "Drawing1.vsdx");
            // get page by name
            Page page = diagram.Pages.GetPage("Page-3");

            foreach (Aspose.Diagram.Shape shape in page.Shapes)
            {
                if (shape.Name == "Process1")
                {
                    Prop property = shape.Props.GetProp("Name1");
                    Console.WriteLine(property.Label.Value + ": " + property.Value.Val);
                }
            }
            //ExEnd:ReadShapePropByName
        }
    }
}
